src/main/java/uk/gov/hmcts/reform/coh/controller/DeadlineController.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.coh.controller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.coh.domain.OnlineHearing;
+import uk.gov.hmcts.reform.coh.service.OnlineHearingService;
+import uk.gov.hmcts.reform.coh.service.QuestionService;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/continuous-online-hearings/{onlineHearingId}")
+public class DeadlineController {
+
+    private static final Logger log = LoggerFactory.getLogger(DeadlineController.class);
+
+    private OnlineHearingService onlineHearingService;
+    private QuestionService questionService;
+
+    @Autowired
+    public DeadlineController(
+        OnlineHearingService onlineHearingService,
+        QuestionService questionService
+    ) {
+        this.onlineHearingService = onlineHearingService;
+        this.questionService = questionService;
+    }
+
+    @PutMapping("/deadline-extensions")
+    public ResponseEntity requestExtensionForQuestion(@PathVariable UUID onlineHearingId) {
+        OnlineHearing onlineHearing = new OnlineHearing();
+        onlineHearing.setOnlineHearingId(onlineHearingId);
+        Optional<OnlineHearing> optionalOnlineHearing = onlineHearingService.retrieveOnlineHearing(onlineHearing);
+
+        if (!optionalOnlineHearing.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Online hearing not found");
```

Comment:
Not convinced that this is the correct HTTP response code. I'd expect to use 404 for the resource that's being request. I suppose it's debatable if it's really the online hearing that's being requested or the question

Reply:
The request concerns all questions of the hearing, and if it doesn't exist... 404 came over as a natural candidate.